app/models/service_template_ansible_playbook.rb

```diff
@@ -1,6 +1,7 @@
 class ServiceTemplateAnsiblePlaybook < ServiceTemplateGeneric
   before_destroy :check_retirement_potential, :prepend => true
   around_destroy :around_destroy_callback, :prepend => true
+  after_rollback :rollback_job_templates, :on => :create
```

Comment:
I've tested the following scenarios without any surprises with the `after_rollback` callback.

Test 1
1. Create single Provision Catalog Item ( with a forced `raise` inside the transaction)
2. Error shows up in the UI, the job_template no longer exists in the EmbeddedAnsible provider

Test 2
1. Create provision / retirement
2. Force a `raise` in the ServiceTemplate creation
3. Error shows up in the UI, the job_templates no longer exist in the EmbeddedAnsible 

Test 3
1. Create provision / retirement with new Dialog. 
2. Force a `raise` in the Dialog creation
3. Error shows up in the UI, the job_templates no longer exist in the EmbeddedAnsible provider

Test 4
1. Edit an existing Provision / Retirement Catalog Item - adding a new dialog item to the provision
2. Force a `raise` in the Dialog Creation
3. Error shows up in the UI - Both playbooks still exist in the EmbeddedProvider. UI shows original Dialog under the Provision tab.

The UI fix [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/1361) resolves the issue of passing in a duplicate dialog name and also was tested during the above examples